In [1]:
import numpy as np
import pandas as pd

# 1. Import datasets

In [2]:
adverse_reactions = pd.read_csv('adverse_reactions.csv')
patients = pd.read_csv('patients.csv')
treatments_cut = pd.read_csv('treatments_cut.csv')
treatments = pd.read_csv('treatments.csv')

# 2. Lets see the datasets

In [3]:
patients

,patient_id,assigned_sex,given_name,surname,address,city,state,zip_code,country,contact,birthdate,weight,height,bmi
0,1,female,Zoe,Wellish,576 Brown Bear Drive,Rancho California,California,92390.0,United States,951-719-9170ZoeWellish@superrito.com,7/10/1976,121.7,66,19.6
1,2,female,Pamela,Hill,2370 University Hill Road,Armstrong,Illinois,61812.0,United States,PamelaSHill@cuvox.de+1 (217) 569-3204,4/3/1967,118.8,66,19.2
2,3,male,Jae,Debord,1493 Poling Farm Road,York,Nebraska,68467.0,United States,402-363-6804JaeMDebord@gustr.com,2/19/1980,177.8,71,24.8
3,4,male,Liêm,Phan,2335 Webster Street,Woodbridge,NJ,7095.0,United States,PhanBaLiem@jourrapide.com+1 (732) 636-8246,7/26/1951,220.9,70,31.7
4,5,male,Tim,Neudorf,1428 Turkey Pen Lane,Dothan,AL,36303.0,United States,334-515-7487TimNeudorf@cuvox.de,2/18/1928,192.3,27,26.1
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
498,499,male,Mustafa,Lindström,2530 Victoria Court,Milton Mills,ME,3852.0,United States,207-477-0579MustafaLindstrom@jourrapide.com,4/10/1959,181.1,72,24.6
499,500,male,Ruman,Bisliev,494 Clarksburg Park Road,Sedona,AZ,86341.0,United States,928-284-4492RumanBisliev@gustr.com,3/26/1948,239.6,70,34.4
500,501,female,Jinke,de Keizer,649 Nutter Street,Overland Park,MO,64110.0,United States,816-223-6007JinkedeKeizer@teleworm.us,1/13/1971,171.2,67,26.8
501,502,female,Chidalu,Onyekaozulu,3652 Boone Crockett Lane,Seattle,WA,98109.0,United States,ChidaluOnyekaozulu@jourrapide.com1 360 443 2060,2/13/1952,176.9,67,27.7


In [4]:
treatments_cut

,given_name,surname,auralin,novodra,hba1c_start,hba1c_end,hba1c_change
0,jožka,resanovič,22u - 30u,-,7.56,7.22,0.34
1,inunnguaq,heilmann,57u - 67u,-,7.85,7.45,NaN
2,alwin,svensson,36u - 39u,-,7.78,7.34,NaN
3,thể,lương,-,61u - 64u,7.64,7.22,0.92
4,amanda,ribeiro,36u - 44u,-,7.85,7.47,0.38
...,...,...,...,...,...,...,...
65,rovzan,kishiev,32u - 37u,-,7.75,7.41,0.34
66,jakob,jakobsen,-,28u - 26u,7.96,7.51,0.95
67,bernd,schneider,48u - 56u,-,7.74,7.44,0.30
68,berta,napolitani,-,42u - 44u,7.68,7.21,NaN


In [5]:
treatments

,given_name,surname,auralin,novodra,hba1c_start,hba1c_end,hba1c_change
0,veronika,jindrová,41u - 48u,-,7.63,7.20,NaN
1,elliot,richardson,-,40u - 45u,7.56,7.09,0.97
2,yukitaka,takenaka,-,39u - 36u,7.68,7.25,NaN
3,skye,gormanston,33u - 36u,-,7.97,7.62,0.35
4,alissa,montez,-,33u - 29u,7.78,7.46,0.32
...,...,...,...,...,...,...,...
275,albina,zetticci,45u - 51u,-,7.93,7.73,0.20
276,john,teichelmann,-,49u - 49u,7.90,7.58,NaN
277,mathea,lillebø,23u - 36u,-,9.04,8.67,0.37
278,vallie,prince,31u - 38u,-,7.64,7.28,0.36


In [6]:
adverse_reactions

,given_name,surname,adverse_reaction
0,berta,napolitani,injection site discomfort
1,lena,baer,hypoglycemia
2,joseph,day,hypoglycemia
3,flavia,fiorentino,cough
4,manouck,wubbels,throat irritation
5,jasmine,sykes,hypoglycemia
6,louise,johnson,hypoglycemia
7,albinca,komavec,hypoglycemia
8,noe,aranda,hypoglycemia
9,sofia,hermansen,injection site discomfort


# About datasets

## 3. Summary of Datasets

The datasets have 4 csv files. The `patients` file contain the patients information. The `treatmennts` file contain the auralin and novodra (iinjection or pills) dose. also contain the hba1c i.e the blood sugar level. then the `treatment_cut` file is same as treatment file. and last file is `adverse_reaction` which contains the sideeffects or bad effects that patient faced.

**So overall this is medical data that has patients data some patients taking novodra and some taking auralin which is the pill or ijection to control blood sugar levels. So the patients taking the dose and the blood sugar level at the start and at the end is mentioned with difference.**

Total patients are 503 and only 350 have taken the auralin or novodra pill/injection. 

In [7]:
treatments[treatments['auralin']=='-'].shape

(143, 7)

In [8]:
treatments_cut[treatments_cut['auralin']=='-'].shape

(32, 7)

There are total 175 patients who took auralin.  
There are total 175 patients who took novodra. 

## Columns descriptions

`patients`  
- `patient_id` - A unique id of patient
- `assigned_sex` - Gender of a patient
- `given_name` - First name of a patient
- `surname` - Last name of a patient
- `address` - Address of a patient
- `city` - City of patient (corresponding to Address)
- `state` - State corresponding to city
- `zip_code` - pin code or zip code corresponding to city
- `country`	- Country
- `contact` - Email and contact number of a patient
- `birthdate` - Date of birth of a patient
- `weight` - Weight of a patient in pounds
- `height` - Height of a patient in inches
- `bmi` - Body mass index of a patient

`treatments` and `treatments_cut`
- `given_name` - Name of a patient refer `patients` table for other info using name+surname
- `surname` - Surname of a patient refer `patients` table for other info using name+surname
- `auralin` - It cintains the unit of taking the auralin. It is a range like 41u-48u. It can have meaning like the patient took auralin started from 41u and at the end of clinical trial.
- `novodra` - Same as above just the drug is different i.e. novodra
- `hba1c_start` - It have the level of hba1c at the start of clinical trial. hba1c is the blood sugar level. 
- `hba1c_end` - It have the level of hba1c at the end of clinical trial. 
- `hba1c_change` - How much differenc is there from start to end in hba1c level.

`adverse_reactions`
- `given_name` - Name of a patient
- `surname` - Surname of patient
- `adverse_reaction` - It contain the adverse reaction the patient noticed. The adverse reaction mean side effect.

### Additional information
This is the patients data. 175 taking auralin and 175 taking novodra. And there hba1c levels at start of clinical test and at the end of it. 

# Assessment

## Issues with the dataset

1. **Dirty Data**
- Table - `Patients`

    - patient_id = 9 has misspelled name 'Dsvid' instead of David `accuracy`
    - state col sometimes contain full name and some times abbrivietation `consistency`
    - zip code col has entries with 4 digit `validity`
    - data missing for 12 patients in address,city,	state,zip_code	,country, contact `completion`
    - incorrect data type assigned to sex, zip code, birthdate `validity`
    - duplicate entries by the name of John Doe `accuracy`
    - one patient has weight = 48 pounds `accuracy`
    - one patient has height = 27 inches `accuracy`

- Table - `Treatments` & `Treatments_cut`

    - given_name and surname col is is all lower case `consistency`
    - remove u from Auralin and Novadra cols `validity`
    - '-' in novadra and Auralin col treated as nan `validity`
    - missing values in hba1c_change col `completion`
    - 1 duplicate entry by the name Joseph day `accuracy`
    - in hba1c_change 9 instead of 4 `accuracy`

- Table - `Adverse_reactions`

    - given_name and surname are all in lower case `consistency`

2. **Messy Data**

- Table - `Patients`

    - contact col contains both phone and email

- Table - `Treatments` & `Treatments_cut`

    - Auralin and Novadra col should be split into 2 cols start and end dose
    - merge both the tables

- Table - `Adverse_reactions`

    - This table should not exist independently

In [9]:
# Manual assessment
# export data to perform the manual assessment

with pd.ExcelWriter('clinical_trials.xlsx') as writer:
  patients.to_excel(writer,sheet_name='patients')
  treatments.to_excel(writer,sheet_name='treatments')
  treatments_cut.to_excel(writer,sheet_name='treatments_cut')
  adverse_reactions.to_excel(writer,sheet_name='adverse_reactions')

In [53]:
# Programatic assessment

head and tail  
sample  
info   
isnull  
duplicated  
describe  

### head, tail and sample

In [20]:
# head and tail
# sample

the manual assessment can be done using these functions.

### info

In [22]:
# info
# In this we can get to know two things - 1. null values, 2. Dtype of each column

In [23]:
patients.info()

<class 'pandas.DataFrame'>
RangeIndex: 503 entries, 0 to 502
Data columns (total 14 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   patient_id    503 non-null    int64  
 1   assigned_sex  503 non-null    str    
 2   given_name    503 non-null    str    
 3   surname       503 non-null    str    
 4   address       491 non-null    str    
 5   city          491 non-null    str    
 6   state         491 non-null    str    
 7   zip_code      491 non-null    float64
 8   country       491 non-null    str    
 9   contact       491 non-null    str    
 10  birthdate     503 non-null    str    
 11  weight        503 non-null    float64
 12  height        503 non-null    int64  
 13  bmi           503 non-null    float64
dtypes: float64(3), int64(2), str(9)
memory usage: 55.1 KB


In [38]:
# The 4, 5, 6, 7, 8 and 9 columns have null values
patients[patients['address'].isnull()]

,patient_id,assigned_sex,given_name,surname,address,city,state,zip_code,country,contact,birthdate,weight,height,bmi
209,210,female,Lalita,Eldarkhanov,NaN,NaN,NaN,NaN,NaN,NaN,8/14/1950,143.4,62,26.2
219,220,male,Mỹ,Quynh,NaN,NaN,NaN,NaN,NaN,NaN,4/9/1978,237.8,69,35.1
230,231,female,Elisabeth,Knudsen,NaN,NaN,NaN,NaN,NaN,NaN,9/23/1976,165.9,63,29.4
234,235,female,Martina,Tománková,NaN,NaN,NaN,NaN,NaN,NaN,4/7/1936,199.5,65,33.2
242,243,male,John,O'Brian,NaN,NaN,NaN,NaN,NaN,NaN,2/25/1957,205.3,74,26.4
249,250,male,Benjamin,Mehler,NaN,NaN,NaN,NaN,NaN,NaN,10/30/1951,146.5,69,21.6
257,258,male,Jin,Kung,NaN,NaN,NaN,NaN,NaN,NaN,5/17/1995,231.7,69,34.2
264,265,female,Wafiyyah,Asfour,NaN,NaN,NaN,NaN,NaN,NaN,11/3/1989,158.6,63,28.1
269,270,female,Flavia,Fiorentino,NaN,NaN,NaN,NaN,NaN,NaN,10/9/1937,175.2,61,33.1
278,279,female,Generosa,Cabán,NaN,NaN,NaN,NaN,NaN,NaN,12/16/1962,124.3,69,18.4


So there are total 12 patients who have not specified the `address`, `city`, `state`,	`zip_code`,	`country` and `contact`.

In [25]:
treatments.info()

<class 'pandas.DataFrame'>
RangeIndex: 280 entries, 0 to 279
Data columns (total 7 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   given_name    280 non-null    str    
 1   surname       280 non-null    str    
 2   auralin       280 non-null    str    
 3   novodra       280 non-null    str    
 4   hba1c_start   280 non-null    float64
 5   hba1c_end     280 non-null    float64
 6   hba1c_change  171 non-null    float64
dtypes: float64(3), str(4)
memory usage: 15.4 KB


In [43]:
# column 6 `hba1c_change` have null values
treatments['hba1c_change'].isnull().sum()

np.int64(109)

In [49]:
treatments[treatments['hba1c_change'].isnull()].head()

,given_name,surname,auralin,novodra,hba1c_start,hba1c_end,hba1c_change
0,veronika,jindrová,41u - 48u,-,7.63,7.20,NaN
2,yukitaka,takenaka,-,39u - 36u,7.68,7.25,NaN
8,saber,ménard,-,54u - 54u,8.08,7.70,NaN
9,asia,woźniak,30u - 36u,-,7.76,7.37,NaN
10,joseph,day,29u - 36u,-,7.70,7.19,NaN


So the column `hba1c_change` have missing values

In [26]:
treatments_cut.info()

<class 'pandas.DataFrame'>
RangeIndex: 70 entries, 0 to 69
Data columns (total 7 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   given_name    70 non-null     str    
 1   surname       70 non-null     str    
 2   auralin       70 non-null     str    
 3   novodra       70 non-null     str    
 4   hba1c_start   70 non-null     float64
 5   hba1c_end     70 non-null     float64
 6   hba1c_change  42 non-null     float64
dtypes: float64(3), str(4)
memory usage: 4.0 KB


In [52]:
treatments_cut['hba1c_change'].isnull().sum()

np.int64(28)

Same as treatments the treatments_cut also have missing vaules in `hba1c_change` column

In [29]:
adverse_reactions.info()

<class 'pandas.DataFrame'>
RangeIndex: 34 entries, 0 to 33
Data columns (total 3 columns):
 #   Column            Non-Null Count  Dtype
---  ------            --------------  -----
 0   given_name        34 non-null     str  
 1   surname           34 non-null     str  
 2   adverse_reaction  34 non-null     str  
dtypes: str(3)
memory usage: 948.0 bytes


All okay

### duplicated

In [55]:
# duplicated
# used to find duplicate records

In [60]:
# Patients
patients[patients.duplicated()]

,patient_id,assigned_sex,given_name,surname,address,city,state,zip_code,country,contact,birthdate,weight,height,bmi


In [64]:
patients[patients[['assigned_sex', 'given_name', 'surname']].duplicated()]

,patient_id,assigned_sex,given_name,surname,address,city,state,zip_code,country,contact,birthdate,weight,height,bmi
229,230,male,John,Doe,123 Main Street,New York,NY,12345.0,United States,johndoe@email.com1234567890,1/1/1975,180.0,72,24.4
237,238,male,John,Doe,123 Main Street,New York,NY,12345.0,United States,johndoe@email.com1234567890,1/1/1975,180.0,72,24.4
244,245,male,John,Doe,123 Main Street,New York,NY,12345.0,United States,johndoe@email.com1234567890,1/1/1975,180.0,72,24.4
251,252,male,John,Doe,123 Main Street,New York,NY,12345.0,United States,johndoe@email.com1234567890,1/1/1975,180.0,72,24.4
277,278,male,John,Doe,123 Main Street,New York,NY,12345.0,United States,johndoe@email.com1234567890,1/1/1975,180.0,72,24.4


In [65]:
patients[patients['address'].duplicated()]

,patient_id,assigned_sex,given_name,surname,address,city,state,zip_code,country,contact,birthdate,weight,height,bmi
29,30,male,Jake,Jakobsen,648 Old Dear Lane,Port Jervis,New York,12771.0,United States,JakobCJakobsen@einrot.com+1 (845) 858-7707,8/1/1985,155.8,67,24.4
219,220,male,Mỹ,Quynh,NaN,NaN,NaN,NaN,NaN,NaN,4/9/1978,237.8,69,35.1
229,230,male,John,Doe,123 Main Street,New York,NY,12345.0,United States,johndoe@email.com1234567890,1/1/1975,180.0,72,24.4
230,231,female,Elisabeth,Knudsen,NaN,NaN,NaN,NaN,NaN,NaN,9/23/1976,165.9,63,29.4
234,235,female,Martina,Tománková,NaN,NaN,NaN,NaN,NaN,NaN,4/7/1936,199.5,65,33.2
237,238,male,John,Doe,123 Main Street,New York,NY,12345.0,United States,johndoe@email.com1234567890,1/1/1975,180.0,72,24.4
242,243,male,John,O'Brian,NaN,NaN,NaN,NaN,NaN,NaN,2/25/1957,205.3,74,26.4
244,245,male,John,Doe,123 Main Street,New York,NY,12345.0,United States,johndoe@email.com1234567890,1/1/1975,180.0,72,24.4
249,250,male,Benjamin,Mehler,NaN,NaN,NaN,NaN,NaN,NaN,10/30/1951,146.5,69,21.6
251,252,male,John,Doe,123 Main Street,New York,NY,12345.0,United States,johndoe@email.com1234567890,1/1/1975,180.0,72,24.4


In [68]:
patients[patients['address']=='648 Old Dear Lane']

,patient_id,assigned_sex,given_name,surname,address,city,state,zip_code,country,contact,birthdate,weight,height,bmi
24,25,male,Jakob,Jakobsen,648 Old Dear Lane,Port Jervis,New York,12771.0,United States,JakobCJakobsen@einrot.com+1 (845) 858-7707,8/1/1985,155.8,67,24.4
29,30,male,Jake,Jakobsen,648 Old Dear Lane,Port Jervis,New York,12771.0,United States,JakobCJakobsen@einrot.com+1 (845) 858-7707,8/1/1985,155.8,67,24.4


In [69]:
patients[patients['address']=='2476 Fulton Street']

,patient_id,assigned_sex,given_name,surname,address,city,state,zip_code,country,contact,birthdate,weight,height,bmi
131,132,female,Sandra,Taylor,2476 Fulton Street,Rainelle,WV,25962.0,United States,304-438-2648SandraCTaylor@dayrep.com,10/23/1960,206.1,64,35.4
282,283,female,Sandy,Taylor,2476 Fulton Street,Rainelle,WV,25962.0,United States,304-438-2648SandraCTaylor@dayrep.com,10/23/1960,206.1,64,35.4


In [70]:
patients[patients['address']=='2778 North Avenue']

,patient_id,assigned_sex,given_name,surname,address,city,state,zip_code,country,contact,birthdate,weight,height,bmi
97,98,male,Patrick,Gersten,2778 North Avenue,Burr,NE,68324.0,United States,PatrickGersten@rhyta.com402-848-4923,5/3/1954,138.2,71,19.3
502,503,male,Pat,Gersten,2778 North Avenue,Burr,Nebraska,68324.0,United States,PatrickGersten@rhyta.com402-848-4923,5/3/1954,138.2,71,19.3


In [74]:
patients[patients['contact'].duplicated()]

,patient_id,assigned_sex,given_name,surname,address,city,state,zip_code,country,contact,birthdate,weight,height,bmi
29,30,male,Jake,Jakobsen,648 Old Dear Lane,Port Jervis,New York,12771.0,United States,JakobCJakobsen@einrot.com+1 (845) 858-7707,8/1/1985,155.8,67,24.4
219,220,male,Mỹ,Quynh,NaN,NaN,NaN,NaN,NaN,NaN,4/9/1978,237.8,69,35.1
229,230,male,John,Doe,123 Main Street,New York,NY,12345.0,United States,johndoe@email.com1234567890,1/1/1975,180.0,72,24.4
230,231,female,Elisabeth,Knudsen,NaN,NaN,NaN,NaN,NaN,NaN,9/23/1976,165.9,63,29.4
234,235,female,Martina,Tománková,NaN,NaN,NaN,NaN,NaN,NaN,4/7/1936,199.5,65,33.2
237,238,male,John,Doe,123 Main Street,New York,NY,12345.0,United States,johndoe@email.com1234567890,1/1/1975,180.0,72,24.4
242,243,male,John,O'Brian,NaN,NaN,NaN,NaN,NaN,NaN,2/25/1957,205.3,74,26.4
244,245,male,John,Doe,123 Main Street,New York,NY,12345.0,United States,johndoe@email.com1234567890,1/1/1975,180.0,72,24.4
249,250,male,Benjamin,Mehler,NaN,NaN,NaN,NaN,NaN,NaN,10/30/1951,146.5,69,21.6
251,252,male,John,Doe,123 Main Street,New York,NY,12345.0,United States,johndoe@email.com1234567890,1/1/1975,180.0,72,24.4


In [75]:
patients[patients['contact']=='JakobCJakobsen@einrot.com+1 (845) 858-7707']

,patient_id,assigned_sex,given_name,surname,address,city,state,zip_code,country,contact,birthdate,weight,height,bmi
24,25,male,Jakob,Jakobsen,648 Old Dear Lane,Port Jervis,New York,12771.0,United States,JakobCJakobsen@einrot.com+1 (845) 858-7707,8/1/1985,155.8,67,24.4
29,30,male,Jake,Jakobsen,648 Old Dear Lane,Port Jervis,New York,12771.0,United States,JakobCJakobsen@einrot.com+1 (845) 858-7707,8/1/1985,155.8,67,24.4


In [76]:
patients[patients['contact']=='304-438-2648SandraCTaylor@dayrep.com']

,patient_id,assigned_sex,given_name,surname,address,city,state,zip_code,country,contact,birthdate,weight,height,bmi
131,132,female,Sandra,Taylor,2476 Fulton Street,Rainelle,WV,25962.0,United States,304-438-2648SandraCTaylor@dayrep.com,10/23/1960,206.1,64,35.4
282,283,female,Sandy,Taylor,2476 Fulton Street,Rainelle,WV,25962.0,United States,304-438-2648SandraCTaylor@dayrep.com,10/23/1960,206.1,64,35.4


In [77]:
patients[patients['contact']=='PatrickGersten@rhyta.com402-848-4923']

,patient_id,assigned_sex,given_name,surname,address,city,state,zip_code,country,contact,birthdate,weight,height,bmi
97,98,male,Patrick,Gersten,2778 North Avenue,Burr,NE,68324.0,United States,PatrickGersten@rhyta.com402-848-4923,5/3/1954,138.2,71,19.3
502,503,male,Pat,Gersten,2778 North Avenue,Burr,Nebraska,68324.0,United States,PatrickGersten@rhyta.com402-848-4923,5/3/1954,138.2,71,19.3


The person named John Doe have duplicate records.  
All other persons who have duplicate address are not the same persons they are maybe relatives.  
All other persons who have duplicate contact are also not same persons so nothing to worry.

In [ ]:
# treatments

In [79]:
treatments[treatments.duplicated()]

,given_name,surname,auralin,novodra,hba1c_start,hba1c_end,hba1c_change
136,joseph,day,29u - 36u,-,7.7,7.19,NaN


In [81]:
treatments[treatments['given_name']=='joseph']

,given_name,surname,auralin,novodra,hba1c_start,hba1c_end,hba1c_change
10,joseph,day,29u - 36u,-,7.7,7.19,NaN
136,joseph,day,29u - 36u,-,7.7,7.19,NaN


In [82]:
treatments[treatments[['given_name', 'surname']].duplicated()]

,given_name,surname,auralin,novodra,hba1c_start,hba1c_end,hba1c_change
136,joseph,day,29u - 36u,-,7.7,7.19,NaN


Have duplicate record of person `joseph	day`

In [ ]:
# treatments_cut

In [84]:
treatments_cut[treatments_cut.duplicated()]

,given_name,surname,auralin,novodra,hba1c_start,hba1c_end,hba1c_change


In [ ]:
# adverse_reactions

In [85]:
adverse_reactions[adverse_reactions.duplicated()]

,given_name,surname,adverse_reaction


No duplicated entry in `treatments_cut` and `adverse_reactions`.

### describe  

In [ ]:
# patients

In [86]:
patients.describe()

,patient_id,zip_code,weight,height,bmi
count,503.000000,491.000000,503.000000,503.000000,503.000000
mean,252.000000,49084.118126,173.434990,66.634195,27.483897
std,145.347859,30265.807442,33.916741,4.411297,5.276438
min,1.000000,1002.000000,48.800000,27.000000,17.100000
25%,126.500000,21920.500000,149.300000,63.000000,23.300000
50%,252.000000,48057.000000,175.300000,67.000000,27.200000
75%,377.500000,75679.000000,199.500000,70.000000,31.750000
max,503.000000,99701.000000,255.900000,79.000000,37.700000


In [90]:
patients[patients['weight']==48.8]

,patient_id,assigned_sex,given_name,surname,address,city,state,zip_code,country,contact,birthdate,weight,height,bmi
210,211,female,Camilla,Zaitseva,4689 Briarhill Lane,Wooster,OH,44691.0,United States,330-202-2145CamillaZaitseva@superrito.com,11/26/1938,48.8,63,19.1


In [91]:
patients[patients['height']==27.000000]

,patient_id,assigned_sex,given_name,surname,address,city,state,zip_code,country,contact,birthdate,weight,height,bmi
4,5,male,Tim,Neudorf,1428 Turkey Pen Lane,Dothan,AL,36303.0,United States,334-515-7487TimNeudorf@cuvox.de,2/18/1928,192.3,27,26.1


- One person have weight 48.8 pound (22.13kg) weight. He has 63inches (5.25feet) height. It can be possible but lets just write it down.
- One person has 27 inches hight and it is not possible cause he has weight 192.3 pounds

In [ ]:
# treatments

In [92]:
treatments.describe()

,hba1c_start,hba1c_end,hba1c_change
count,280.000000,280.000000,171.000000
mean,7.985929,7.589286,0.546023
std,0.568638,0.569672,0.279555
min,7.500000,7.010000,0.200000
25%,7.660000,7.270000,0.340000
50%,7.800000,7.420000,0.380000
75%,7.970000,7.570000,0.920000
max,9.950000,9.580000,0.990000


In [94]:
treatments.sort_values('hba1c_start')

,given_name,surname,auralin,novodra,hba1c_start,hba1c_end,hba1c_change
270,mika,martinsson,34u - 43u,-,7.50,7.17,0.33
113,kari,laatikainen,39u - 43u,-,7.50,7.11,NaN
126,jowita,wiśniewska,-,22u - 23u,7.50,7.08,0.92
53,nasser,mansour,-,33u - 31u,7.51,7.06,0.95
105,finlay,sheppard,-,31u - 30u,7.51,7.17,0.34
...,...,...,...,...,...,...,...
25,benoît,bonami,-,44u - 43u,9.82,9.40,0.92
171,justyna,kowalczyk,24u - 34u,-,9.84,9.44,NaN
81,robert,wagner,43u - 49u,-,9.84,9.52,0.32
75,mackenzie,mckay,-,44u - 43u,9.87,9.48,0.39


In [95]:
treatments.sort_values('hba1c_end')

,given_name,surname,auralin,novodra,hba1c_start,hba1c_end,hba1c_change
39,krisztina,magyar,-,32u - 27u,7.52,7.01,0.51
80,hideki,haraguchi,-,37u - 35u,7.59,7.05,0.54
187,león,reynoso,-,38u - 40u,7.59,7.06,0.53
53,nasser,mansour,-,33u - 31u,7.51,7.06,0.95
86,ananías,enríquez,-,44u - 45u,7.58,7.07,0.51
...,...,...,...,...,...,...,...
192,valur,bjarkason,-,31u - 36u,9.71,9.41,0.30
171,justyna,kowalczyk,24u - 34u,-,9.84,9.44,NaN
75,mackenzie,mckay,-,44u - 43u,9.87,9.48,0.39
81,robert,wagner,43u - 49u,-,9.84,9.52,0.32


In [97]:
treatments.sort_values('hba1c_change', na_position='first')

,given_name,surname,auralin,novodra,hba1c_start,hba1c_end,hba1c_change
0,veronika,jindrová,41u - 48u,-,7.63,7.20,NaN
2,yukitaka,takenaka,-,39u - 36u,7.68,7.25,NaN
8,saber,ménard,-,54u - 54u,8.08,7.70,NaN
9,asia,woźniak,30u - 36u,-,7.76,7.37,NaN
10,joseph,day,29u - 36u,-,7.70,7.19,NaN
...,...,...,...,...,...,...,...
49,jackson,addison,-,42u - 42u,7.99,7.51,0.98
17,gina,cain,-,36u - 36u,7.88,7.40,0.98
32,laura,ehrlichmann,-,43u - 40u,7.95,7.46,0.99
245,wu,sung,-,47u - 48u,7.61,7.12,0.99


In [110]:
treatments.sample()

,given_name,surname,auralin,novodra,hba1c_start,hba1c_end,hba1c_change
88,yumena,nakayama,-,34u - 32u,7.76,7.35,0.91


- column `hba1c_change` it has some values in that the 4 is replaced with 9.

In [ ]:
# treatments_cut

In [111]:
treatments_cut.describe()

,hba1c_start,hba1c_end,hba1c_change
count,70.000000,70.000000,42.000000
mean,7.838000,7.443143,0.518810
std,0.423007,0.418706,0.270719
min,7.510000,7.020000,0.280000
25%,7.640000,7.232500,0.340000
50%,7.730000,7.345000,0.370000
75%,7.860000,7.467500,0.907500
max,9.910000,9.460000,0.970000


In [114]:
treatments_cut.sample(5)

,given_name,surname,auralin,novodra,hba1c_start,hba1c_end,hba1c_change
63,žarka,rap,35u - 48u,-,7.54,7.15,NaN
19,noe,aranda,26u - 34u,-,7.51,7.17,0.34
46,anja,mueller,43u - 56u,-,7.58,7.29,0.29
41,dalmacia,madrid,-,26u - 23u,7.67,7.21,0.96
21,david,villadsen,26u - 39u,-,7.86,7.40,NaN


- Have same problem like `treatments`

In [ ]:
# adverse_reactions

In [115]:
adverse_reactions.describe()

,given_name,surname,adverse_reaction
count,34,34,34
unique,34,33,6
top,berta,johnson,hypoglycemia
freq,1,2,19


There is no numeric column so all okay.